docusaurus/docs/Android/04-compose/07-guides/06-implementing-own-capabilities.mdx

```diff
@@ -0,0 +1,233 @@
+
+```kotlin
+public data class MessageComposerState(
+        val inputValue: String = "",
+        val attachments: List<Attachment> = emptyList(),
+        val action: MessageAction? = null,
+        val validationErrors: List<ValidationError> = emptyList(),
+        val mentionSuggestions: List<User> = emptyList(),
+        val commandSuggestions: List<Command> = emptyList(),
+        val coolDownTime: Int = 0,
+        val messageMode: MessageMode = MessageMode.Normal,
+        val alsoSendToChannel: Boolean = false,
+        val ownCapabilities: Set<String> = setOf()
+)
+```
+
+`MessageComposer` regulates the following capabilities:
+
+- send-message
+- send-links
+- upload-file
+
+### SelectedMessageMenu
+
+```kotlin
+@Composable
+public fun SelectedMessageMenu(
+        message: Message,
+        messageOptions: List<MessageOptionItemState>,
+        ownCapabilities: Set<String>,
+        onMessageAction: (MessageAction) -> Unit,
+        onShowMoreReactionsSelected: () -> Unit,
+        modifier: Modifier = Modifier,
+        shape: Shape = ChatTheme.shapes.bottomSheet,
+        overlayColor: Color = ChatTheme.colors.overlay,
+        reactionTypes: Map<String, ReactionIcon> = ChatTheme.reactionIconFactory.createReactionIcons(),
+        @DrawableRes showMoreReactionsIcon: Int = R.drawable.stream_compose_ic_more,
+        onDismiss: () -> Unit = {},
+        headerContent: @Composable ColumnScope.() -> Unit = {
+            // Default content, uses own capabilities to regulate sending reactions
+        },
+        centerContent: @Composable ColumnScope.() -> Unit = {
+            // Default content
+        },
+)
+```
+
+The file containing `SelectedMessageMenu` can be
+found [here](https://github.com/GetStream/stream-chat-android/blob/0f1f54e912baf7a7284d9a4d1435fb6743aa9c78/stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/components/selectedmessage/SelectedMessageMenu.kt#L60)
+
+`SelectedMessageMenu` regulates the following capability:
+
+- send-reaction
+
+Additionally, you can create a list of `messageOptions` using the following function provided by the SDK:
+
+```kotlin
+@Composable
+public fun defaultMessageOptionsState(
+        selectedMessage: Message,
+        currentUser: User?,
+        isInThread: Boolean,
+        ownCapabilities: Set<String>,
+): List<MessageOptionItemState>
+```
+
+This function uses `ownCapabilities` to produce the correct list of `MessageOptionItemState`.
+
+`defaultMessageOptionsState` regulates the following capabilities:
+
+- send-reply
+- pin-message
+- delete-any-message
+- delete-own-message
+- update-any-message
+- update-own-message
+
+### SelectedReactionsMenu
+
+```kotlin
+@Composable
+public fun SelectedReactionsMenu(
+        message: Message,
```

Comment:
More here

Reply:
Fixed indentation and removed unnecessary parameters.